### PR TITLE
Generate a batch script on Windows (launches exe)

### PR DESCRIPTION
**Pull request by mahge**
- We now generate a <ModelName>.bat file along with other generated files.

    The script updates the PATH with the directories extracted from the model
    and then launches the executable.

  - The APIs `simulate`, `optimize` and `linearize` now launch executables
    using this bat file.

  - OMEdit manually parses the bat file, extracts the additional paths
    needed and updates the PATH itself before launching the executable.
    If it fails to find or parse the bat file a warning message will be printed
    but simulation will be attempted anyway.

  - OMNotebook and OMShell are supposed to use the API functions and
    should be correct as well.

### CODE REVIEW

File: OMEdit/OMEditLIB/Simulation/SimulationOutputWidget.cpp [hunk 1]
```diff
@@ -602,9 +633,13 @@ void SimulationOutputWidget::runSimulationExecutable()
...
 #if defined(_WIN32)
+  QProcessEnvironment processEnvironment = StringHandler::simulationProcessEnvironment();
+
+  QString pathsFromModel = getPathsFromBatFile(fileName + ".bat");
+  processEnvironment.insert("PATH", pathsFromModel + ";" + processEnvironment.value("PATH"));
+
   fileName = fileName.append(".exe");
   QFileInfo fileInfo(mSimulationOptions.getFileName());
-  QProcessEnvironment processEnvironment = StringHandler::simulationProcessEnvironment();
   processEnvironment.insert("PATH", fileInfo.absoluteDir().absolutePath() + ";" + processEnvironment.value("PATH"));
```
Comment (adeas31) [inline OMEdit/OMEditLIB/Simulation/SimulationOutputWidget.cpp:643]: You can set the PATH at once instead of doing it in 2 different lines.

Comment (mahge) [reply OMEdit/OMEditLIB/Simulation/SimulationOutputWidget.cpp:643]: Sure thing. I will update it.

**Comment by adrpo** [comment]
How do you return the exit code of the exe you are calling when you exit the bat file? I did not see that anywhere. You would need to do something like:

```
@echo off
set ERRORLEVEL=
call Model.exe
exit /b %ERRORLEVEL%
```

In general I would have liked to avoid using bat files to run models as all kinds of issues pop up if you have that indirection.
I would have just dumped a Model_setLibraryPaths.cmd and read that into OMEdit and OMC to set the paths before running system or CreateProcess. 

But let's try it this way for now as it is a bit more straightforward if one would need to distribute the executable.

**Comment by mahge** [comment]
> How do you return the exit code of the exe you are calling when you exit the bat file? I did not see that anywhere.

Launching the executable is the last action in the bat file. It should exit with the return code of the executable. Right? It worked for me at least. My knowledge of bat files not so good.

If we add more logic to the file it might be useful to catch and exit with the %ERRORLEVEL% of the executable. I can add it if you want.

> In general I would have liked to avoid using bat files to run models as all kinds of issues pop up if you have that indirection.
I would have just dumped a Model_setLibraryPaths.cmd and read that into OMEdit and OMC to set the paths before running system or CreateProcess.

OMEdit right now works almost as you suggest here. It parses the bat file manually and then sets the paths itself. It does not actually execute the bat file. 

The API functions used to work by internally (inside omc) setting the path after #8552. It is now changed after the suggestions there. We can of course write just the PATHs to a cmd and then execute that before launching the executable.  

> But let's try it this way for now as it is a bit more straightforward if one would need to distribute the executable.

Sounds good. As a bonus we can now also directly run simulations on Windows command lines from anywhere with just one command (without having to copy even the OpenModelica dlls).

**Comment by mahge** [comment]
> How do you return the exit code of the exe you are calling when you exit the bat file? I did not see that anywhere.

It is now updated. The generated file should look something like this:

```cmd
@echo off
set PATH=.....;%PATH%;
set ERRORLEVEL=
call "%CD%/ModelName.exe"
set RESULT=%ERRORLEVEL%

exit /b %RESULT%
```

If at some point we need to do cleanup or something similar it can go in between the `call` and exit.

**Comment by adrpo** [comment]
Re-running this with MinGW label to build it on windows as well.

**Review by adrpo — APPROVED** [review]
Looks fine to me but we will need to test:
- paths with spaces in them (might need to be quoted)
- paths with special chars (might not work)

to make sure these work fine.

**Pull request merged**